test/integration/multicluster/tgt/target_test.go

```diff
@@ -20,12 +23,18 @@ var (
 	httpReqRE = regexp.MustCompile(
 		`request_total\{direction="outbound",target_addr="[0-9\.]+:8080",target_ip="[0-9\.]+",target_port="[0-9\.]+",tls="true",server_id="default\.multicluster-statefulset\.serviceaccount\.identity\.linkerd\.cluster\.local",dst_control_plane_ns="linkerd",dst_namespace="multicluster-statefulset",dst_pod="nginx-statefulset-0",dst_serviceaccount="default",dst_statefulset="nginx-statefulset"\} [1-9]\d*`,
 	)
-	dgCmd = []string{"diagnostics", "proxy-metrics", "--namespace",
+	dgCmd = []string{"--context=" + targetCtx, "diagnostics", "proxy-metrics", "--namespace",
 		"linkerd-multicluster", "deploy/linkerd-gateway"}
 )
 
 func TestMain(m *testing.M) {
 	TestHelper = testutil.NewTestHelper()
+	// Block test execution until viz extension (last extension to be installed)
+	// is running
+	TestHelper.WaitUntilDeployReady(testutil.LinkerdVizDeployReplicas)
```

Comment:
The context here will be the environment's, which isn't necessarily `target`

Reply:
Yeah that's true. I added a new method on the TestHelper to wait for deploy readiness with a context. Behind the scenes, it'll attempt to switch the context to the given argument and then call `WaitUntilDeployReady` with the given deployment spec.